app/frontend/wallet/cardano-wallet.js

```diff
@@ -81,10 +78,23 @@ const CardanoWallet = async (options) => {
     return await cryptoProvider.getWalletId()
   }
 
-  async function prepareTx(address, coins) {
-    const txAux = await prepareTxAux(address, coins)
+  async function prepareSignedTx(address, coins) {
+    const txAux = await prepareTxAux(address, coins).catch((e) => {
+      debugLog(e)
+      throw NamedError('TransactionCorrupted')
+    })
+    console.log(txAux.inputs)
+    const transactions = []
+    for (const input of txAux.inputs) {
```

Comment:
this for cycle can be refactored to:
```
const rawInputTxs = await Promise.all(
    txAux.inputs.map(
        ({txHash}) => blockChainExplorer.fetchTxRaw(txHash)
    )
)
```
moreover, this has the benefit of fetching the raw transactions in parallel